fdb-extensions/src/test/java/com/apple/foundationdb/test/FDBTestEnvironment.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * FDBTestEnvironment.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.test;
+
+import org.assertj.core.api.Assumptions;
+import org.yaml.snakeyaml.Yaml;
+
+import javax.annotation.Nullable;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
```

Comment:
I thought about putting this in `testFixtures`, but since other projects already depend on the tests configuration, doing so would require moving all the other classes into a `testFixtures`, and I figured that would be better as a separate PR.

Reply:
I made a change in main to move them into a test fixture, and then went to test publishing of `testFixtures` so that `yaml-tests` could be used by clients for their tests, but it looks like we can't publish `testFixtures`: https://github.com/FoundationDB/fdb-record-layer/commit/e9db5c04ba449c64d846ca389879ffae56c6d730#diff-9515853bb3a0c957c759e9c453a9e17941f4333751715e75d74efed681fadac0R60 I'm going to look into creating a `test-utils` library....